fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -6670,7 +6663,7 @@ public Money[] retriveIncomeOutstandingTillDate(final LocalDate paymentDate) {
                     paidFromFutureInstallments = paidFromFutureInstallments.plus(balancesForCurrentPeroid[4])
                             .minus(balancesForCurrentPeroid[2]);
                 }
-            } else if (installment.getDueDate().isAfter(paymentDate)) {
```

Comment:
Are you sure?

Reply:
Yes, I am sure. 
Before this else branch we checked these conditions already:
!DateUtils.isBefore(paymentDate, installment.getDueDate())
DateUtils.isAfter(paymentDate, installment.getFromDate())
so this check was redundant and not needed:
DateUtils.isBefore(paymentDate, installment.getDueDate())